crates/notedeck/src/time.rs

```diff
@@ -1,55 +1,324 @@
+use crate::tr;
 use std::time::{SystemTime, UNIX_EPOCH};
 
-pub fn time_ago_since(timestamp: u64) -> String {
-    let now = SystemTime::now()
-        .duration_since(UNIX_EPOCH)
-        .expect("Time went backwards")
-        .as_secs();
+// Time duration constants in seconds
+const ONE_MINUTE_IN_SECONDS: u64 = 60;
+const ONE_HOUR_IN_SECONDS: u64 = 3600;
+const ONE_DAY_IN_SECONDS: u64 = 86_400;
+const ONE_WEEK_IN_SECONDS: u64 = 604_800;
+const ONE_MONTH_IN_SECONDS: u64 = 2_592_000; // 30 days
+const ONE_YEAR_IN_SECONDS: u64 = 31_536_000; // 365 days
+
+// Range boundary constants for match patterns
+const MAX_SECONDS: u64 = ONE_MINUTE_IN_SECONDS - 1;
+const MAX_SECONDS_FOR_MINUTES: u64 = ONE_HOUR_IN_SECONDS - 1;
+const MAX_SECONDS_FOR_HOURS: u64 = ONE_DAY_IN_SECONDS - 1;
+const MAX_SECONDS_FOR_DAYS: u64 = ONE_WEEK_IN_SECONDS - 1;
+const MAX_SECONDS_FOR_WEEKS: u64 = ONE_MONTH_IN_SECONDS - 1;
+const MAX_SECONDS_FOR_MONTHS: u64 = ONE_YEAR_IN_SECONDS - 1;
 
+/// Calculate relative time between two timestamps
+fn time_ago_between(timestamp: u64, now: u64) -> String {
     // Determine if the timestamp is in the future or the past
     let duration = if now >= timestamp {
         now.saturating_sub(timestamp)
     } else {
         timestamp.saturating_sub(now)
     };
 
-    let future = timestamp > now;
-    let relstr = if future { "+" } else { "" };
+    let time_str = match duration {
+        0..=2 => tr!(
+            "now",
+            "Relative time for very recent events (less than 3 seconds)"
+        ),
+        3..=MAX_SECONDS => tr!("{count}s", "Relative time in seconds", count = duration),
+        ONE_MINUTE_IN_SECONDS..=MAX_SECONDS_FOR_MINUTES => tr!(
+            "{count}m",
+            "Relative time in minutes",
+            count = duration / ONE_MINUTE_IN_SECONDS
+        ),
+        ONE_HOUR_IN_SECONDS..=MAX_SECONDS_FOR_HOURS => tr!(
+            "{count}h",
+            "Relative time in hours",
+            count = duration / ONE_HOUR_IN_SECONDS
+        ),
+        ONE_DAY_IN_SECONDS..=MAX_SECONDS_FOR_DAYS => tr!(
+            "{count}d",
+            "Relative time in days",
+            count = duration / ONE_DAY_IN_SECONDS
+        ),
+        ONE_WEEK_IN_SECONDS..=MAX_SECONDS_FOR_WEEKS => tr!(
+            "{count}w",
+            "Relative time in weeks",
+            count = duration / ONE_WEEK_IN_SECONDS
+        ),
+        ONE_MONTH_IN_SECONDS..=MAX_SECONDS_FOR_MONTHS => tr!(
+            "{count}mo",
+            "Relative time in months",
+            count = duration / ONE_MONTH_IN_SECONDS
+        ),
+        _ => tr!(
+            "{count}y",
+            "Relative time in years",
+            count = duration / ONE_YEAR_IN_SECONDS
+        ),
+    };
 
-    let years = duration / 31_536_000; // seconds in a year
-    if years >= 1 {
-        return format!("{}{}yr", relstr, years);
```

Comment:
Note that this and other time abbreviations are not the globally recognized standard for US English. I replaced them with more standard ones expected for US English (copying the format from Damus iOS for inspiration which does it right).